appmap/pytest.py

```diff
@@ -1,20 +1,59 @@
 import pytest
 
-import appmap.unittest
 from appmap._implementation import testing_framework
+import appmap
+import appmap.wrapt as wrapt
+
+class recorded_testcase:
+    def __init__(self, item):
+        self.item = item
+
+    @wrapt.decorator
+    def __call__(self, wrapped, _, args, kwargs):
+        item = self.item
+        with item.session.appmap.record(
+                item.cls,
+                item.name,
+                method_id=item.originalname,
+                location=item.location):
+            wrapped(*args, **kwargs)
 
 
 if appmap.enabled():
     @pytest.hookimpl
     def pytest_sessionstart(session):
-        """Set up AppMap testing session and associate it with pytest and unittest sessions."""
-        # Use the same session for unittest in case there are any unittest.TestCases.
-        appmap.unittest.session = session.appmap = testing_framework.session(name='pytest', version=pytest.__version__)
+        session.appmap = testing_framework.session(name='pytest', version=pytest.__version__)
+
+    @pytest.hookimpl(hookwrapper=True)
+    def pytest_runtest_call(item):
+        # The presence of a `_testcase` attribute on an item indicates
+        # that it was created from a `unittest.TestCase`. An item
+        # created from a TestCase has an `_obj` attribute, assigned
+        # during in setup, which holds the actual test
+        # function. Wrapping that function will capture the recording
+        # we want. `_obj` gets unset during teardown, so there's no
+        # chance of rewrapping it.
+        #
+        # However, depending on the user's configuration, `item._obj`
+        # may have been already instrumented for recording. In this
+        # case, it will be a `wrapt` class, rather than just a
+        # function. This is fine: the decorator we apply here will be
+        # called first, starting the recording. Next, the
+        # instrumentation decorator will be called, recording the
+        # `call` event. Finally, the original function will be called,
+        # running the test case. (This nesting of function calls is
+        # verified by the expected appmap in the test for a unittest
+        # TestCase run by pytest.)
+        if hasattr(item, '_testcase'):
+            item.obj = recorded_testcase(item)(item.obj)
```

Comment:
The comment mentions `_obj`, however, `obj` is wrapped here. Is this a mistake in the comment or in the code? If the latter, are we missing a test here?

Reply:
I'm not sure. I'll look at it again and make sure everything lines up.

Reply:
It was a mistake in the comment. I've fixed it.